tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,170 @@
+    pass
+
+
+# Monkey patching connect method
+def mock_connect_soc(self, *args, **kwargs):
+    pass
+
+
+# Monkey patching publish method
+def mock_publish(self, *args, **kwargs):
+    pass
+
+
+# Monkey patching subscribe method
+def mock_subscribe(self, *args, **kwargs):
+    pass
+
+
+class MqttDccCommsTest(unittest.TestCase):
+    """
+    Unit test cases for MQTT DccComms
+    """
+
+    def setUp(self):
+        """
+        Setup all required parameters for MQTT DCC communication tests
+        :return:
+        """
+        # Monkey patched MQTT connect method
+        Mqtt.connect_soc = mock_connect_soc
+
+        # Create Edge System
+        self.edge_system = Dell5KEdgeSystem(config['EdgeSystemName'])
+
+        self.url = config['BrokerIP']
+        self.port = config['BrokerPort']
+        self.clean_session = True
+        self.publish_message = "test-message"
+
+        # Creating messaging attributes
+        self.mqtt_msg_attr = MqttMessagingAttributes(pub_topic="publish-topic")
+
+        # Instantiate client for DCC communication
+        self.client = MqttDccComms(edge_system_name=self.edge_system.name,
+                                   url=self.url, port=self.port, clean_session=self.clean_session)
+
+    def test_init_no_msgattr(self):
+        """
+        Test MqttDccComms initialisation with:
+        #1. Empty client ID
+        #2. Non-empty client ID
+        :return:
+        """
+        Mqtt.connect_soc = mock_connect_soc
+
+        # Test with empty client ID
+        self.assertIsNotNone(MqttDccComms(edge_system_name=self.edge_system.name,
+                                          url=self.url, port=self.port,
+                                          clean_session=self.clean_session))
+
+        # Test with non-empty client ID
+        self.assertIsNotNone(MqttDccComms(edge_system_name=self.edge_system.name,
+                                          url=self.url, port=self.port,
+                                          client_id="unit-test", clean_session=self.clean_session))
+
+    def test_init_with_msgattr(self):
+        """
+        Test MqttDccComms initialisation with:
+        #1. Valid messaging attributes
+        #2. Invalid messaging attributes
+        :return:
+        """
+        Mqtt.connect_soc = mock_connect_soc
+
+        # Test with valid messaging attributes
+        self.assertIsNotNone(MqttDccComms(edge_system_name=self.edge_system.name,
+                                          url=self.url, port=self.port, clean_session=self.clean_session,
+                                          mqtt_msg_attr=self.mqtt_msg_attr))
+
+        # Test with invalid messaging attributes
+        self.assertRaises(TypeError, lambda: MqttDccComms(edge_system_name=self.edge_system.name,
+                                                          url=self.url, port=self.port,
+                                                          clean_session=self.clean_session, mqtt_msg_attr=""))
+
```

Comment:
Please add test cases for:
1) client_id (This is very important, as we had a bug previously)
- auto-generated
- manually passed by user
2) _connect()
- Ensure all params are correctly passed down to transport's connect by mocking transport's connect()